### PR TITLE
added pyro5 recipe

### PR DESCRIPTION
**Pull request by irmen**
Checklist

- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [x] Source is from official source
- [x] Package does not vend other packages
- [x] Build number is 0
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your
      recipe (see
      [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos)
      for more details)
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

"I'm willing to be listed here"  ;-)

### CODE REVIEW

**Comment by irmen** [comment]
@whart222 @Juanlu001  sorry to bother you but perhaps you can help me. I'm trying to make a conda forge recipe for Pyro5.  In the past, you helped me with the recipe for Pyro4.  I'm facing some difficulties getting the Pyro5 recipe to build... :( 
the CI complains about "RuntimeError: Setuptools downloading is disabled in conda build. Be sure to add all dependencies in the meta.yaml"  but I think I did exactly that?  Also, I don't think this was ever a problem with the Pyro4 recipe?

**Comment by Juanlu001** [comment]
@irmen Hi! I am _almost_ sure it's because of `setup_requires` and `test_require`. These use `easy_install` under the hood and are not recommended anymore. Notice that [`setup.py test` is deprecated](https://github.com/pypa/setuptools/issues/1684) therefore, `test_requires` does not make sense anymore, and `setup_requires` has been replaced by [PEP 518 and `pyproject.toml`](https://www.python.org/dev/peps/pep-0518/).

**Comment by irmen** [comment]
@Juanlu001 thanks for your time to have a look. I've never heard of pep 518 until now, and certainly haven't implemented that in Pyro4 so I assume Pyro5 could work without it as well.
I'll remove the test_require .  Also I'm assuming I have to create a new Pyro5 release from that to retry this build recipe, correct?

**Comment by Juanlu001** [comment]
> Also I'm assuming I have to create a new Pyro5 release from that to retry this build recipe, correct?

Correct!

**Comment by irmen** [comment]
@Juanlu001 oh well look at that, the build is green!  Thank you so much!  :tada:

File: recipes/pyro5/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  entry_points:
+    - pyro5-ns = Pyro5.nameserver:main
+    - pyro5-nsc = Pyro5.nsc:main
+    - pyro5-echoserver = Pyro5.utils.echoserver:main
+    - pyro5-check-config = Pyro5.configure:dump
+    - pyro5-httpgateway = Pyro5.utils.httpgateway:main
+
+
+requirements:
+  build:
```
Comment (marcelotrevisani) [inline recipes/pyro5/meta.yaml]: Just ``host`` should be enough, could you please remove the ``build``?

File: recipes/pyro5/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+  version: {{ version }}
+
+source:
+  fn: Pyro5-{{ version }}.tar.gz
+  url: https://pypi.io/packages/source/P/Pyro5/Pyro5-{{ version }}.tar.gz
+  sha256: fe7998b0bfab7e6f1bf02fa7d69e4b4c53305da745bae07d604eeea139316828
+
+build:
+  number: 0
+  skip: True  # [py<35]
```
Comment (marcelotrevisani) [inline recipes/pyro5/meta.yaml]: I think this recipe can be ``noarch: python``. You can select the ``python`` version on ``host`` and ``run`` requirements

File: recipes/pyro5/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+
+requirements:
+  build:
+    - python
+    - setuptools
+
+  host:
+    - pip
+    - python
```
Comment (marcelotrevisani) [inline recipes/pyro5/meta.yaml]:
```suggestion
    - python >=3
```

File: recipes/pyro5/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    - setuptools
+    - serpent >=1.27
+    
+  run:
+    - python
```
Comment (marcelotrevisani) [inline recipes/pyro5/meta.yaml]:
```suggestion
    - python >=3.0
```

File: recipes/pyro5/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,60 @@
+{% set version = "5.5.1" %}
+
+package:
+  name: pyro5
+  version: {{ version }}
+
+source:
+  fn: Pyro5-{{ version }}.tar.gz
```
Comment (marcelotrevisani) [inline recipes/pyro5/meta.yaml]: This ``fn`` is not necessary, could you please remove it?
```suggestion
```

File: recipes/pyro5/meta.yaml (hunk 1 above)
Comment (irmen) [reply recipes/pyro5/meta.yaml]: ok

File: recipes/pyro5/meta.yaml (hunk 2 above)
Comment (irmen) [reply recipes/pyro5/meta.yaml]: Hm, I started with that, and then it went on to build it with python 2.7 which failed. Pyro5 is for python 3.5+   Is there something else I should have set perhaps?

Comment (dbast) [reply recipes/pyro5/meta.yaml]: See the other suggestions, they explain it.

Comment (irmen) [reply recipes/pyro5/meta.yaml]: ok I will try!

**Comment by irmen** [comment]
thank you for merging

**Pull request merged**